src/pages/settings/PasswordPage.js

```diff
@@ -160,17 +223,18 @@ class PasswordPage extends Component {
                                 autoCompleteType="password"
                                 textContentType="password"
                                 value={this.state.confirmNewPassword}
-                                onChangeText={confirmNewPassword => this.setState({confirmNewPassword})}
-                                onSubmitEditing={this.handleChangePassword}
-                                onBlur={() => this.onBlurConfirmPassword()}
+                                onChangeText={text => this.clearErrorAndSetValue('confirmNewPassword', text)}
+                                hasError={this.state.errors.confirmNewPassword}
+                                errorText={this.getErrorText('confirmNewPassword')}
+                                onSubmitEditing={this.validateAndSubmitForm}
                             />
                         </View>
-                        {!this.state.shouldShowPasswordConfirmError && !isEmpty(this.props.account.error) && (
+                        {!this.state.errors.confirmPasswordMatch && !_.isEmpty(this.props.account.error) && (
                             <Text style={styles.formError}>
                                 {this.props.account.error}
                             </Text>
                         )}
-                        {this.state.shouldShowPasswordConfirmError && (
+                        {this.state.errors.confirmPasswordMatch && (
```

Comment:
Should we still be showing this error? It feels kind of out of place and I'm not sure it's correct.

If the first password is valid but the confirmation password doesn't match then the password confirmation field is the one with the "error"? Seems like it wouldn't make sense to show this:

![2021-10-18_09-42-58](https://user-images.githubusercontent.com/32969087/137796387-5f200c2f-d729-4563-baa0-5402a8affd0c.png)

Reply:
I've updated. I was a bit confused on best practices for tagging multiple error flags to a single field. I've added an approach. Let me know if that works for you.